fdbcli/TenantCommands.actor.cpp

```diff
@@ -444,60 +503,75 @@ ACTOR Future<bool> tenantGetCommand(Reference<IDatabase> db, std::vector<StringR
-				doc.get("lock_state", tenantLockState);
-
-				bool hasTenantState = doc.tryGet("tenant_state", tenantState);
-				bool hasLockId = doc.tryGet("lock_id", lockId);
-				bool hasTenantGroup = doc.tryGet("tenant_group.printable", tenantGroup);
-				bool hasAssignedCluster = doc.tryGet("assigned_cluster.printable", assignedCluster);
-				bool hasError = doc.tryGet("error", error);
-
-				fmt::print("  id: {}\n", id);
-				fmt::print("  prefix: {}\n", printable(prefix));
-				if (hasTenantState) {
-					fmt::print("  tenant state: {}\n", printable(tenantState));
+			tenantGetCmdOutput(jsonObject, useJson);
+			return true;
+		} catch (Error& e) {
+			try {
+				wait(safeThreadFutureToFuture(tr->onError(e)));
+			} catch (Error& finalErr) {
+				state std::string errorStr;
+				if (finalErr.code() == error_code_special_keys_api_failure) {
+					std::string str = wait(getSpecialKeysFailureErrorMessage(tr));
+					errorStr = str;
+				} else if (useJson) {
+					errorStr = finalErr.what();
+				} else {
+					throw finalErr;
 				}
 
-				fmt::print("  lock state: {}\n", tenantLockState);
-				if (hasLockId) {
-					fmt::print("  lock id: {}\n", lockId);
+				if (useJson) {
+					json_spirit::mObject resultObj;
+					resultObj["type"] = "error";
+					resultObj["error"] = errorStr;
+					fmt::print(
+					    "{}\n",
+					    json_spirit::write_string(json_spirit::mValue(resultObj), json_spirit::pretty_print).c_str());
+				} else {
+					fmt::print(stderr, "ERROR: {}\n", errorStr.c_str());
 				}
 
-				if (hasTenantGroup) {
-					fmt::print("  tenant group: {}\n", tenantGroup);
-				}
-				if (hasAssignedCluster) {
-					fmt::print("  assigned cluster: {}\n", printable(assignedCluster));
-				}
-				if (hasError) {
-					fmt::print("  error: {}\n", error);
-				}
+				return false;
+			}
+		}
+	}
+}
+
+// tenant get command
+ACTOR Future<bool> tenantGetIdCommand(Reference<IDatabase> db, std::vector<StringRef> tokens) {
+	if (tokens.size() < 3 || tokens.size() > 4 || (tokens.size() == 4 && tokens[3] != "JSON"_sr)) {
+		fmt::print("Usage: tenant getId <ID> [JSON]\n\n");
+		fmt::print("Prints metadata associated with the given tenant ID.\n");
+		fmt::print("If JSON is specified, then the output will be in JSON format.\n");
+		return false;
+	}
+
+	state bool useJson = tokens.size() == 4;
+	state int64_t tenantId;
+	int n = 0;
+	if (sscanf(tokens[2].toString().c_str(), "%" PRId64 "%n", &tenantId, &n) != 1 || n != tokens[2].size() ||
+	    tenantId < 0) {
+		fmt::print(stderr, "ERROR: invalid ID `{}'\n", tokens[2].toString().c_str());
+		return false;
+	}
+	state Reference<ITransaction> tr = db->createTransaction();
+
+	loop {
+		try {
+			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+			state ClusterType clusterType = wait(TenantAPI::getClusterType(tr));
+			state std::string tenantJson;
+			if (clusterType != ClusterType::METACLUSTER_MANAGEMENT) {
+				fmt::print(stderr, "ERROR: get by ID should only be run on a management cluster.\n");
```

Comment:
Rather than disallow reading by ID on a standalone/data cluster, we could forgo use of the special keys and use `TenantAPI::getTenantTransaction`. Eventually we may want to switch that to using special keys, but that doesn't have to be now.

Reply:
I can make this change, but I believe this would mean `tenant get t1` and `tenant getId 1` would return different results for the same tenant. Is that fine?

Reply:
What would be different about them?

Reply:
In standalone clusters, reading from the special keys and parsing out the json output does not include the actual tenant name. `TenantMapEntry::toJson()` will include the name field

Reply:
The special keys command is using the same `toJson` function, so it should be including the tenant name. If it isn't then I think we should add it so that it matches the metacluster functionality.

Reply:
I was mistaken about this, I think they should match, so I'm asserting on this condition for the fdbcli tests now